compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -236,6 +236,27 @@ public boolean isAssignable(BType source, BType target) {
         return isAssignable(source, target, new ArrayList<>());
     }
 
+    /**
+     * Recursively checks whether the given literal can be assigned to the given type.
+     *
+     * @param target  target type
+     * @param literal literal to check
+     * @return true if the literal can be assigned to the type, false otherwise.
+     */
+    public boolean isAssignable(BType target, BLangLiteral literal) {
+        if (target.tag == TypeTags.FINITE) {
+            return isAssignableToFiniteType(target, literal);
+        }
+        if (target.tag == TypeTags.UNION) {
+            for (BType memberType : ((BUnionType) target).memberTypes) {
+                if (isAssignable(memberType, literal)) {
+                    return true;
+                }
+            }
+        }
+        return false;
```

Comment:
Can we check this? because this returns false if target type is not finite or union. with the given method name, it's kind of misleading.